manifests/params.pp

```diff
@@ -142,7 +142,12 @@
       $bindir               = pick($bindir, "/usr/lib/postgresql/${version}/bin")
       $datadir              = pick($datadir, "/var/lib/postgresql/${version}/main")
       $confdir              = pick($confdir, "/etc/postgresql/${version}/main")
-      $service_status       = pick($service_status, "/etc/init.d/${service_name} status | /bin/egrep -q 'Running clusters: .+|online'")
+      if versioncmp($::operatingsystemrelease, '8.0') >= 0 {
+        # Jessie uses systemd
+        $service_status = pick($service_status, "/usr/sbin/service ${service_name}@*-main status")
+      } else {
+        $service_status = pick($service_status, "/etc/init.d/${service_name} status | /bin/egrep -q 'Running clusters: .+|online'")
```

Comment:
Ah right, sorry for that.
On 18 Mar 2015 18:38, "Morgan Haskel" <notifications@github.com> wrote:

> In manifests/params.pp
> <https://github.com/puppetlabs/puppetlabs-postgresql/pull/586#discussion_r26687209>
> :
>
> > @@ -142,7 +142,12 @@
> >        $bindir               = pick($bindir, "/usr/lib/postgresql/${version}/bin")
> >        $datadir              = pick($datadir, "/var/lib/postgresql/${version}/main")
> >        $confdir              = pick($confdir, "/etc/postgresql/${version}/main")
> > -      $service_status       = pick($service_status, "/etc/init.d/${service_name} status | /bin/egrep -q 'Running clusters: .+|online'")
> > +      if versioncmp($::operatingsystemrelease, '8.0') >= 0 {
> > +        # Jessie uses systemd
> > +        $service_status = pick($service_status, "/usr/sbin/service ${service_name}@*-main status")
> > +      } else {
> > +        $service_status = pick($service_status, "/etc/init.d/${service_name} status | /bin/egrep -q 'Running clusters: .+|online'")
>
> this definitely failed in acceptance tests on ubuntu 14.04. thus #588
> <https://github.com/puppetlabs/puppetlabs-postgresql/pull/588>
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/puppetlabs/puppetlabs-postgresql/pull/586/files#r26687209>
> .
>